test-network/organizations/cfssl/registerEnroll.sh

```diff
@@ -0,0 +1,276 @@
+
+    fi
+
+    if [[ $TYPE == "orderer" ]]; then
+        generate_orderer_certs $CERT_DIR "$USER"
+    fi
+
+    if [[ $TYPE == "admin" ]]; then
+        generate_user_certs "$CERT_DIR" "$USER" "$TYPE"
+    fi
+
+    find . -name "*.csr" -print0 | xargs -0 rm
+
+}
+
+function generate_user_certs() {
+
+    CERT_DIR=$1
+    USER=$2
+    TYPE=$3
+
+    mkdir -p $CERT_DIR/users/$USER/tls
+
+    for DIR in cacerts keystore signcerts tlscacerts; do
+        mkdir -p $CERT_DIR/users/$USER/msp/$DIR
+    done
+
+    sed -e "s/{USER}/$USER/g" <"$PWD/organizations/cfssl/${TYPE}-csr-template.json" >$PWD/organizations/cfssl/${TYPE}-${USER}-csr.json
+
+    cfssl gencert \
+        -ca=$CERT_DIR/ca/ca.pem \
+        -ca-key=$CERT_DIR/ca/ca-key.pem \
+        -config=$PWD/organizations/cfssl/cert-signing-config.json \
+        -cn="$USER" \
+        -hostname="$USER,localhost,127.0.0.1" \
+        -profile="sign" \
+        $PWD/organizations/cfssl/${TYPE}-${USER}-csr.json | cfssljson -bare $CERT_DIR/users/$USER/msp/signcerts/cert
+
+    mv $CERT_DIR/users/$USER/msp/signcerts/cert-key.pem $CERT_DIR/users/$USER/msp/keystore/cert-key.pem
+    cp $CERT_DIR/ca/ca.pem $CERT_DIR/users/$USER/msp/cacerts
+    cp $CERT_DIR/ca/ca.pem $CERT_DIR/users/$USER/msp/tlscacerts
+
+    echo 'NodeOUs:
+    Enable: true
+    ClientOUIdentifier:
+      Certificate: cacerts/ca.pem
+      OrganizationalUnitIdentifier: client
+    PeerOUIdentifier:
+      Certificate: cacerts/ca.pem
+      OrganizationalUnitIdentifier: peer
+    AdminOUIdentifier:
+      Certificate: cacerts/ca.pem
+      OrganizationalUnitIdentifier: admin
+    OrdererOUIdentifier:
+      Certificate: cacerts/ca.pem
+      OrganizationalUnitIdentifier: orderer' >$CERT_DIR/users/$USER/msp/config.yaml
+
+    cfssl gencert \
+        -ca=$CERT_DIR/ca/ca.pem \
+        -ca-key=$CERT_DIR/ca/ca-key.pem \
+        -config=$PWD/organizations/cfssl/cert-signing-config.json \
+        -cn="$USER" \
+        -hostname="$USER,localhost,127.0.0.1" \
+        -profile="tls" \
+        $PWD/organizations/cfssl/${TYPE}-${USER}-csr.json | cfssljson -bare $CERT_DIR/users/$USER/tls/client
+
+    cp $CERT_DIR/ca/ca.pem $CERT_DIR/users/$USER/tls/ca.crt
+    mv $CERT_DIR/users/$USER/tls/client-key.pem $CERT_DIR/users/$USER/tls/client.key
+    mv $CERT_DIR/users/$USER/tls/client.pem $CERT_DIR/users/$USER/tls/client.crt
+
+    rm $PWD/organizations/cfssl/${TYPE}-${USER}-csr.json
+
+}
+
+function generate_peer_certs() {
+    CERT_DIR=$1
+    USER=$2
+
+    for DIR in cacerts keystore signcerts tlscacerts; do
+        mkdir -p "$CERT_DIR/peers/$USER/msp/$DIR"
+    done
+
+    mkdir -p "$CERT_DIR/peers/$USER/tls"
+    sed -e "s/{USER}/$USER/g" <"$PWD/organizations/cfssl/peer-csr-template.json" >"$PWD/organizations/cfssl/peer-${USER}.json"
+
+    cfssl gencert \
+        -ca="$CERT_DIR/ca/ca.pem" \
+        -ca-key="$CERT_DIR/ca/ca-key.pem" \
+        -config="$PWD/organizations/cfssl/cert-signing-config.jso"n \
```

Comment:
Somehow the generation worked even with this typo, but I assume it needs to be fixed.